### PR TITLE
Adds the ability to remove monkey/xeno skins and make costumes with them

### PR DESCRIPTION
**Pull request by Firecage**
This PR adds the ability to skin monkeys and xenos using the kitchen spike. You simply put them on the spike like you always do, but instead of immediately harvesting meat, you will first harvest the skin/hide and then the meat.

Using the hides, you can make monkey/xeno masks(monkey)/helmets(Xeno) and suits. For the former it takes 1 hide, for the latter 2, so in total 3. Not that much for the monkeys, but it would take work to get 3 xeno hides for a full xeno costume.

Note, you can also harvest xeno hides from simple animal xenos by butchering them.

All sprites used was already in the code. Basically the hides and the costumes.

Tested everything and everything still works properly.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 1]
```diff
@@ -347,7 +350,7 @@
 		else
 			user << "<span class='notice'> [src] is dead, medical items won't bring it back to life.</span>"
 			return
-	if(meat_type && (stat == DEAD))	//if the animal has a meat, and if it is dead.
+	if(meat_type && (stat == DEAD) || skin_type && (stat == DEAD))	//if the animal has a meat, and if it is dead.
```
Comment (Miauw62) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: should be ```if((meat_type || skin_type) && stat == DEAD)```

File: code/game/objects/structures/kitchen_spike.dm [hunk 2]
```diff
@@ -23,6 +25,8 @@
 			src.occupied = 1
 			src.meat = 5
 			src.meattype = 1
+			src.skin = 1
+			src.skintype = 1
```
Comment (Miauw62) [inline code/game/objects/structures/kitchen_spike.dm]: pls no magic numbers

File: code/game/objects/structures/kitchen_spike.dm [hunk 3]
```diff
@@ -55,10 +68,14 @@
 	if(..())
 		return
 	if(src.occupied)
-		if(src.meattype == 1)
-			if(src.meat > 1)
+		if(src.meattype == MEATTYPE_MONKEY && src.skintype == SKINTYPE_MONKEY)
+			if(src.skin == 1)
```
Comment (Miauw62) [inline code/game/objects/structures/kitchen_spike.dm]: this should be >=, not ==.

File: code/game/objects/structures/kitchen_spike.dm [hunk 4]
```diff
@@ -66,10 +83,14 @@
 				usr << "You remove the last piece of meat from the monkey!"
 				src.icon_state = "spike"
 				src.occupied = 0
-		else if(src.meattype == 2)
-			if(src.meat > 1)
+		else if(src.meattype == MEATTYPE_ALIEN && src.skintype == SKINTYPE_ALIEN)
+			if(src.skin == 1)
```
Comment (Miauw62) [inline code/game/objects/structures/kitchen_spike.dm]: same here.

File: code/game/objects/structures/kitchen_spike.dm (hunk 3 above)
Comment (RemieRichards) [inline code/game/objects/structures/kitchen_spike.dm]: actually skin is only ever 1 or 0, so this should be ```if(skin)```

File: code/game/objects/structures/kitchen_spike.dm (hunk 4 above)
Comment (RemieRichards) [inline code/game/objects/structures/kitchen_spike.dm]: ```if(skin)```

Comment (Miauw62) [inline code/game/objects/structures/kitchen_spike.dm]: no, if it's supposed to be like meat it is the amount of skin left, so it should be >

Comment (RemieRichards) [inline code/game/objects/structures/kitchen_spike.dm]: a creature literally only has 1 skin, and every usage of the variable is either 0 or 1.

File: code/game/objects/structures/kitchen_spike.dm (hunk 3 above)
Comment (Firecage) [inline code/game/objects/structures/kitchen_spike.dm]: Mmmm, actually, Miauw is right. If an admin edits the spiked monkey to have more than 1 skin, then it should be >=

Comment (RemieRichards) [inline code/game/objects/structures/kitchen_spike.dm]: yeah but realistically it shouldn't be more than 1, but if design wise you want to go that way, then fine.

**Comment by Lumipharon** [comment]
Monkey fur coat/boats/fur lined cloak when?

**Comment by Firecage** [comment]
Lumipharon. When someone sprites those things.

**Comment by Miauw62** [comment]
it works :+1:

**Pull request merged**